path_planning_plugins/src/path_planning_plugins.cpp

```diff
@@ -0,0 +1,259 @@
+#include <geometry_msgs/PoseArray.h>
+#include <godel_msgs/PathPlanning.h>
+#include "path_planning_plugins.h"
+#include <pluginlib/class_list_macros.h>
+#include <ros/node_handle.h>
+#include <tf/transform_datatypes.h>
+
+
+const static std::string DEFAULT_PARAM_PREFIX = "/path_planning_params/";
+const static std::string DISCRETIZATION = DEFAULT_PARAM_PREFIX + "discretization";
+const static std::string MARGIN = DEFAULT_PARAM_PREFIX + "margin";
+const static std::string OVERLAP = DEFAULT_PARAM_PREFIX + "overlap";
+const static std::string SAFE_TRAVERSE_HEIGHT = DEFAULT_PARAM_PREFIX + "safe_traverse_height";
+const static std::string SCAN_WIDTH = DEFAULT_PARAM_PREFIX + "scan_width";
+const static std::string TOOL_RADIUS = DEFAULT_PARAM_PREFIX + "tool_radius";
+const static double MIN_BOUNDARY_LENGTH = 0.1; // 10 cm
+const static std::string PATH_GENERATION_SERVICE = "process_path_generator";
+
+// Together these constants define a 5cm approach and departure path for the laser scans
+const static int SCAN_APPROACH_STEP_COUNT = 5;
+const static double SCAN_APPROACH_STEP_DISTANCE = 0.01; // 1cm
+
+namespace path_planning_plugins
+{
+  bool Openveronoi::populatePlanningParameters(ros::NodeHandle& nh, path_planning_base::PlanningParams& params)
+  {
+    try
+    {
+      nh.getParam(DISCRETIZATION, params.discretization);
+      nh.getParam(MARGIN, params.margin);
+      nh.getParam(OVERLAP, params.overlap);
+      nh.getParam(SAFE_TRAVERSE_HEIGHT, params.traverse_height);
+      nh.getParam(SCAN_WIDTH, params.scan_width);
+      nh.getParam(TOOL_RADIUS, params.tool_radius);
+      return true;
+    }
+    catch(const std::exception& e)
+    {
+      ROS_ERROR_STREAM("Unable to populate path planning parameters" << e.what());
+    }
+    return false;
+  }
+
+  static godel_process_path::PolygonBoundaryCollection
+  filterPolygonBoundaries(const godel_process_path::PolygonBoundaryCollection& boundaries)
+  {
+    godel_process_path::PolygonBoundaryCollection filtered_boundaries;
+
+    for (std::size_t i = 0; i < boundaries.size(); ++i)
+    {
+      const godel_process_path::PolygonBoundary& bnd = boundaries[i];
+      double circ = godel_process_path::polygon_utils::circumference(bnd);
+
+      if (circ < MIN_BOUNDARY_LENGTH)
+      {
+        ROS_WARN_STREAM("Ignoring boundary with length " << circ);
+      }
+      else if (!godel_process_path::polygon_utils::checkBoundary(bnd))
+      {
+        ROS_WARN_STREAM("Ignoring ill-formed boundary");
+      }
+      else
+      {
+        filtered_boundaries.push_back(bnd);
+        godel_process_path::polygon_utils::filter(filtered_boundaries.back(), 0.1);
+        std::reverse(filtered_boundaries.back().begin(), filtered_boundaries.back().end());
+      }
+    }
+    return filtered_boundaries;
+  }
+
+
+
+  // Blend Planner
+
+  void Openveronoi::BlendPlanner::init(pcl::PolygonMesh mesh)
+  {
+    mesh_ = mesh;
+    mesh_importer_ = new mesh_importer::MeshImporter(false);
```

Comment:
On reflection, I'm not sure mesh_importer_ needs to be a private member. We really just use it to calculate boundaries and a pose. Once we get to the end of generatePath, there's no reason to hold on to it. I'll move it to generatePath and use a unique_ptr.